### PR TITLE
feat(accumulators.jl) : add constructor to Accumulator

### LINKED ISSUE #573: Accumulator has non-parseable `repr`
```julia
julia> using DataStructures

julia> x = [1, 1, 2, 2, 3];

julia> counter(x)
Accumulator{Int64,Int64} with 3 entries:
  2 => 2
  3 => 1
  1 => 2

julia> repr(counter(x))
"Accumulator(2 => 2,3 => 1,1 => 2)"

julia> Accumulator(2 => 2,3 => 1,1 => 2)
ERROR: MethodError: no method matching Accumulator(::Pair{Int64,Int64}, ::Pair{Int64,Int64}, ::Pair{Int64,Int64})
```
the valid constructor using `Pair`s is 
```julia
julia> Accumulator(Dict(2 => 2,3 => 1,1 => 2))
Accumulator{Int64,Int64} with 3 entries:
  2 => 2
  3 => 1
  1 => 2
```

Also the printing is missing a space after `,`:
```julia
(2 => 1,1 => 1)
```
should be
```julia
(2 => 1, 1 => 1)

### PR DESCRIPTION
**Pull request by karan0299**
fixes #573 
Made `Accumulator(2 => 2, 3 => 1, 1 => 2)` a valid constructor by accepting variable number of pairs as inputs and making dictionary out of them.

### CODE REVIEW

**Comment by oxinabox** [comment]
thanks for this.
Please add a test of the changed behavour which would have failed before

**Comment by karan0299** [comment]
@oxinabox Whenever you get a chance, this PR is ready for review.
Thanks!

**Comment by karan0299** [comment]
I cant figure out why travis-ci is showing error.
The exact error is `The command "julia --project=benchmark -e ' using Pkg; Pkg.instantiate(); include("benchmark/runjudge.jl");'" failed and exited with 1 during`

File: src/accumulator.jl [hunk 1]
```diff
@@ -235,3 +236,18 @@ function Base.intersect!(a::Accumulator, b::Accumulator)
...
+        print(io,"Accumulator{$T,$V}(")
+    end
+    for (count, (k, v)) in enumerate(acc)
+        print(io, k, " => ", v)
+        if count < l
+            print(io, ", ")
+        end
+    end
+    print(io, ")")
+end
```
Comment (oxinabox) [inline src/accumulator.jl]: Is this change needed?

**Comment by oxinabox** [comment]
https://github.com/JuliaCollections/DataStructures.jl/pull/581#issuecomment-590463982

I suggest running that command locally and seeing what is up

File: src/accumulator.jl (hunk 1 above)
Comment (karan0299) [reply src/accumulator.jl]: Yes I think , since a very important point  is mentioned in the issue regarding the space after pairs are printed i.e.
 `(3 => 2,2  => 1)` printed as `(3 => 2, 2 => 1)`.
Also I think it is better to have Overridden Base.show method for it might be useful in future operations.

Comment (oxinabox) [reply src/accumulator.jl]: Fair, thanks

Comment (oxinabox) [reply src/accumulator.jl]: Fair, thanks

**Comment by karan0299** [comment]
> [#581 (comment)](https://github.com/JuliaCollections/DataStructures.jl/pull/581#issuecomment-590463982)
> 
> I suggest running that command locally and seeing what is up

@oxinabox 
Yeah, I tried locally running the command but encountering no such error. In fact everything is running smoothly.
I will be obliged if you help me out here.
Thanks.

**Comment by oxinabox** [comment]
hmm it runs fine locally for me also

**Comment by oxinabox** [comment]
looks like it was just travis having a blip, I restarted the job and it looks like it is running fine now

**Comment by karan0299** [comment]
Thanks! :smile:

**Comment by oxinabox** [comment]
thanks
